### PR TITLE
feat(Objective start button) : Now start button of objectives in scheduleScreen works perfectly

### LINKED ISSUE #183: Implement the Start button logic of the objectives and add study materials
##** Description**
Objective's start button was empty now must be bring you to the activty screen where you can find the subject pdfs

##** Checklist**

- [x] start button must take you to a objective activity screen
- [x] must be able to return to schedule with the back button
- [x] create a completed button when you finish the objective
- [x] objective status on schedule must be updated when the objective is completed

### PR DESCRIPTION
**Pull request by khlilinfo**
# Add CourseExercises Screen Navigation from Daily Objectives

### What problem are we solving?
Users had no way to actually **start working** on their daily objectives. The "Start" button existed but didn't lead anywhere - there was no screen to view course materials, complete exercises, or mark objectives as done.

### Brief background
The daily objectives section showed a list of tasks with "Start" buttons, but clicking them did nothing. Users needed a dedicated workspace to:
- Access course PDFs and exercise materials
- Track their progress on the objective
- Mark the objective as completed when finished

---

### Core Feature: Start Button → CourseExercises Screen Flow

**1. New CourseExercises Screen** (`CourseExercisesRoute.kt`)
   - Displays the selected objective's details (title, course, time estimate)
   - Two tabs: **Course** and **Exercises** for viewing respective PDF materials
   - **Back button**: Returns to Schedule without completing
   - **"Mark as completed" FAB**: Completes the objective and returns to Schedule

**2. Navigation Flow Implementation**
   - Clicking "Start" on any objective navigates to CourseExercises screen
   - Only the CourseExercises UI is visible (Schedule UI completely hidden - no double navigation bars)
   - Completing or canceling returns user to exactly where they left off in Schedule

**3. Objective Completion Logic**
   - "Mark as completed" button calls `objectivesVm.markObjectiveCompleted(objective)`
   - Objective state updates in repository and syncs to Firestore
   - Start button disappears, replaced with "Completed" pill
   - WeekDots automatically update when all day's objectives are complete

**4. Visual Design**
   - Modern glass-effect surfaces matching app theme
   - Course/Exercises tabs with Material icons (LibraryBooks, Description)
   - Time estimate chip (e.g., "30 min focus")
   - Tip text to guide users
   - Gradient "Mark as completed" FAB

### Schema Changes
**Objective model** (existing, no changes):
```kotlin
data class Objective(
    val title: String,
    val course: String,
    val estimateMinutes: Int,
    val completed: Boolean,  // ← Updated when user completes
    val day: DayOfWeek
)
```

**Firestore**: Objectives collection updated when user marks complete (existing sync logic, no schema changes)

### Test Tags Added
Created `CourseExercisesTestTags.kt` with 17 test tags for comprehensive UI testing

---

### Key Decisions & Trade-offs

**1. Conditional Rendering vs. Navigation Component**
- **Decision**: Use conditional rendering (`if (activeObjective != null)`) with early return
- **Why**: 
  - Simpler than setting up NavHost for single modal-like screen
  - Preserves Schedule state (tab selection, scroll position) when returning
  - Instant back navigation (no NavHost animation overhead)
- **Trade-off**: Tighter coupling between Schedule and CourseExercises, but better UX

**2. SharedFlow for Navigation Events**
- **Decision**: ViewModel emits navigation events via SharedFlow; UI collects in LaunchedEffect
- **Why**: Clean separation - ViewModel doesn't know about navigation implementation
- **How it works**:
  ```kotlin
  // ViewModel
  fun startObjective(index: Int) {
    _navigationEvents.emit(ToCourseExercises(objectives[index]))
  }
  
  // UI
  LaunchedEffect(viewModel) {
    viewModel.navigationEvents.collect { event -> onNavigate(event) }
  }
  ```

**3. PDF Placeholders**
- **Decision**: PDF open buttons call empty lambdas (TODOs)
- **Why**: PDF viewing strategy TBD (Android Intent, embedded viewer, web browser)
- **Future work**: Wire up actual PDF opening logic based on project requirements

### Concurrency/Data Consistency
- **Repository updates**: `markObjectiveCompleted()` calls `repository.updateObjective()`
  - FirestoreObjectivesRepository updates Firestore atomically
  - UI reacts to state changes via StateFlow
- **No race conditions**: Navigation state (`activeObjectiveForCourse`) is single-source-of-truth in ScheduleScreen
- **Thread-safe**: All repository operations use coroutines with proper dispatchers

### Performance Considerations
- **Early return optimization**: When CourseExercises is active, entire Schedule Scaffold is skipped (no wasted rendering)
- **Tab switching**: `remember { mutableStateOf(0) }` prevents unnecessary recomposition
- **Lazy loading**: Only render active tab content (Course or Exercises PDF card)

---

### Unit Tests (16)
**File**: `ObjectivesViewModelTest.kt`

**Coverage**:
- ✅ Initial state loads from repository
- ✅ Refresh reloads objectives
- ✅ CRUD operations (add, update, remove, move, setObjectives)
- ✅ Toggle `showWhy` flag
- ✅ Mark objective as completed (updates repository and state)
- ✅ Check if day objectives completed (for WeekDots integration)
- ✅ **Navigation events**: `startObjective(index)` emits `ToCourseExercises(objective)`
- ✅ Edge cases: invalid index, empty list, multiple sequential operations

**Key test**:
```kotlin
@Test
fun startObjective_emits_toCourseExercises_event() = runTest {
  viewModel.navigationEvents.test {
    viewModel.startObjective(0)
    val event = awaitItem()
    assertTrue(event is ObjectiveNavigation.ToCourseExercises)
    assertEquals("Finish Quiz 3", (event as ObjectiveNavigation.ToCourseExercises).objective.title)
  }
}
```

### UI Tests - Isolated (14)
**File**: `CourseExercisesRouteTest.kt`

**Coverage**:
- ✅ Screen renders with all UI elements
- ✅ Back button triggers `onBack()` callback
- ✅ "Mark as completed" FAB triggers `onCompleted()` callback
- ✅ Tab switching between Course and Exercises
- ✅ PDF cards display correct labels and trigger callbacks
- ✅ Estimate chip shows/hides based on `estimateMinutes`
- ✅ All callbacks can be invoked independently
- ✅ Multiple tab toggles work correctly
- ✅ Long titles are ellipsized properly

**Key test**:
```kotlin
@Test
fun completedFab_callsOnCompletedWhenClicked() {
  var completedClicked = false
  
  composeTestRule.setContent {
    CourseExercisesRoute(
        objective = testObjective,
        onCompleted = { completedClicked = true },
        // ... other params
    )
  }
  
  composeTestRule.onNodeWithTag(CourseExercisesTestTags.COMPLETED_FAB).performClick()
  assertEquals(true, completedClicked)
}
```

### UI Tests - Integration (19)
**File**: `ScheduleScreenCourseExercisesNavigationTest.kt`

**Coverage**:

**Basic Schedule UI** (verify no regressions):
- ✅ Schedule renders without crashing
- ✅ Tab row, FAB, objectives section exist
- ✅ Can switch between Day/Week/Month tabs
- ✅ State persists across tab switches
- ✅ Objectives load and display correctly
- ✅ Start buttons exist and are clickable

**Complete Navigation Flow** (the main feature):
- ✅ **Clicking "Start" navigates to CourseExercises**
- ✅ **Schedule UI completely hidden** (tab row, FAB gone - no double navigation bars)
- ✅ **CourseExercises displays correct objective info** (title, course, estimate)
- ✅ **Back button returns to Schedule** (objective remains incomplete)
- ✅ **"Mark as completed" returns to Schedule AND marks objective complete**
- ✅ Tab row exists in CourseExercises (Course/Exercises tabs work)

**Key integration test**:
```kotlin
@Test
fun courseExercises_completedButton_marksObjectiveComplete() {
  resetObjectives()
  composeTestRule.setContent { ScheduleScreen() }
  
  // 1. Navigate to CourseExercises
  composeTestRule.onNodeWithTag(START_BUTTON_0).performScrollTo().performClick()
  composeTestRule.waitUntil { CourseExercises screen appears }
  
  // 2. Click "Mark as completed"
  composeTestRule.onNodeWithTag(COMPLETED_FAB).performClick()
  composeTestRule.waitUntil { Schedule screen reappears }
  
  // 3. Verify objective is completed (Start button gone)
  composeTestRule.onNodeWithTag(START_BUTTON_0).assertDoesNotExist()  ✅
}
```

**Test isolation strategy**:
```kotlin
// Reset repository state manually in each test (not @Before, which causes JUnit4 failure)
private fun resetObjectives() = runBlocking {
  FakeObjectivesRepository.setObjectives(
      listOf(
          Objective("Finish Quiz 3", "CS101", 30, completed = false, DayOfWeek.MONDAY),
          Objective("Outline lab report", "CS101", 20, completed = false, DayOfWeek.TUESDAY),
          Objective("Review 15 flashcards", "ENG200", 10, completed = false, DayOfWeek.WEDNESDAY),
      ))
}
```

### Test Execution
All tests pass reliably:
```bash
# ViewModel tests
✅ 16/16 passed - ObjectivesViewModelTest

# Isolated UI tests  
✅ 14/14 passed - CourseExercisesRouteTest

# Integration tests
✅ 19/19 passed - ScheduleScreenCourseExercisesNavigationTest

Total: 49 tests passing
```

---

### User Flow Demo
1. **Schedule Screen** - User sees "Finish Quiz 3" with Start button
2. **[Tap Start]**
3. **CourseExercises Screen** - Shows objective details, Course tab selected
4. **[Switch to Exercises tab]** - Shows exercises PDF card
5. **[Tap "Mark as completed" FAB]**
6. **Back to Schedule** - "Finish Quiz 3" now shows "Completed" pill, Start button gone

image
image

---

### Functionality
- ✅ Start button navigates to CourseExercises
- ✅ CourseExercises shows objective details
- ✅ Course/Exercises tabs switch correctly
- ✅ Back button returns without completing
- ✅ Mark as completed updates objective and returns
- ✅ Completed objectives show "Completed" pill instead of Start button
- ✅ No double navigation bars

### Code Quality
- ✅ No compile errors
- ✅ Follows existing code style and architecture
- ✅ Proper separation: UI/ViewModel/Repository
- ✅ Test tags follow naming convention

### Testing
- ✅ All unit tests pass (16)
- ✅ All UI tests pass (14)
- ✅ All integration tests pass (19)
- ✅ Tests isolated (no shared state failures)
- ✅ Edge cases covered

### Dependencies
- ✅ Turbine 1.0.0 added for Flow testing
- ✅ Extended Material icons already present
- ✅ No breaking dependency changes

### Documentation
- ✅ Test tags documented
- ✅ Comments in complex logic
- ✅ TODO markers for future work (PDF opening)

---

### Future Enhancements (Out of Scope)
- Wire up actual PDF viewer (Intent or embedded)
- Add Quiz screen navigation (ObjectiveNavigation.ToQuiz)
- Add Resume screen navigation (ObjectiveNavigation.ToResume)
- Auto-advance to next objective after completion
- Track time spent on each objective

---

## Issue Reference

Closes #183

### CODE REVIEW

File: app/src/main/java/com/android/sample/feature/weeks/ui/CourseExercisesRoute.kt [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                        primaryActionLabel = "Open exercises",
+                        onClick = onOpenExercisesPdf,
+                        cardTag = CourseExercisesTestTags.EXERCISES_PDF_CARD)
+              }
+
+              Spacer(Modifier.height(24.dp))
+
+              Text(
+                  text =
+                      "Tip: you can come back and continue later. Once you feel done, tap \"Mark as completed\".",
```
Comment (SamAbd11805) [inline app/src/main/java/com/android/sample/feature/weeks/ui/CourseExercisesRoute.kt:175]: Please extract hardcoded strings into string resources for localization and consistency

File: app/src/main/java/com/android/sample/feature/weeks/ui/CourseExercisesRoute.kt [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                        },
+                        modifier = Modifier.testTag(CourseExercisesTestTags.EXERCISES_TAB))
+                  }
+
+              Spacer(Modifier.height(16.dp))
+
+              when (selectedTab) {
+                0 ->
+                    PdfCard(
+                        title = "Course PDF",
```
Comment (SamAbd11805) [inline app/src/main/java/com/android/sample/feature/weeks/ui/CourseExercisesRoute.kt:157]: Tab labels are hardcoded; move them to string resources to avoid duplication and enable translation.

File: app/src/main/java/com/android/sample/feature/weeks/ui/DailyObjectivesSection.kt [hunk 3]
```diff
@@ -190,3 +204,23 @@ private fun StartButton(onClick: () -> Unit, tag: String? = null) {
         Text("Start", fontWeight = FontWeight.SemiBold)
       }
 }
+
+@Composable
+private fun CompletedPill() {
+  val cs = MaterialTheme.colorScheme
+  Surface(
+      color = cs.primary.copy(alpha = 0.12f),
```
Comment (SamAbd11805) [inline app/src/main/java/com/android/sample/feature/weeks/ui/DailyObjectivesSection.kt:212]: Consider moving the 0.12f alpha value to a theme constant for better consistency across the UI.

File: app/src/main/java/com/android/sample/feature/weeks/viewmodel/ObjectivesViewModel.kt [hunk 4]
```diff
@@ -32,6 +43,10 @@ class ObjectivesViewModel(
   private val _uiState = MutableStateFlow(ObjectivesUiState())
   val uiState = _uiState.asStateFlow()
 
+  // One-shot navigation events the UI can observe.
+  private val _navigationEvents = MutableSharedFlow<ObjectiveNavigation>()
+  val navigationEvents = _navigationEvents.asSharedFlow()
```
Comment (SamAbd11805) [inline app/src/main/java/com/android/sample/feature/weeks/viewmodel/ObjectivesViewModel.kt:48]: Good encapsulation, but consider exposing this as StateFlow or a dedicated UiEvent sealed class for consistency with other screens

**Review by SamAbd11805 — APPROVED** [review]
Great work on this feature — the structure is clean, the UI state flows are consistent, and the navigation events are well isolated. Overall the implementation fits nicely with the existing architecture.

I did notice a few lint/Sonar issues flagged on CourseExercisesRoute.kt (e.g., using mutableIntStateOf instead of mutableStateOf, and some deprecated APIs around L100/L139). Since you mentioned these are already planned to be fixed, all good on my side — just make sure to apply the corrections in your next pass.

Besides that, nothing blocking. The code is readable, the ViewModel logic is safe, and the Composables follow our design conventions.
